src/algebraic_geometry/presheafed_space.lean

```diff
@@ -0,0 +1,173 @@
+-- Copyright (c) 2019 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison
+import category_theory.instances.Top.presheaf
+
+universes v u
+
+open category_theory
+open category_theory.instances
+open category_theory.instances.Top
+open topological_space
+
+variables (C : Type u) [𝒞 : category.{v+1} C]
+include 𝒞
+
+namespace algebraic_geometry
+
+structure PresheafedSpace :=
+(X : Top.{v})
+(𝒪 : X.presheaf C)
+
+variables {C}
+
+namespace PresheafedSpace
+
+instance : has_coe_to_sort (PresheafedSpace.{v} C) :=
+{ S := Type v, coe := λ F, F.X.α }
+
+instance (F : PresheafedSpace.{v} C) : topological_space F := F.X.str
+
+structure hom (F G : PresheafedSpace.{v} C) :=
+(f : F.X ⟶ G.X)
+(c : G.𝒪 ⟶ f _* F.𝒪)
+
+@[extensionality] lemma ext {F G : PresheafedSpace.{v} C} (α β : hom F G)
+  (w : α.f = β.f) (h : α.c ≫ (whisker_right (nat_trans.op (opens.map_iso _ _ w).inv) F.𝒪) = β.c) :
+  α = β :=
+begin
+  cases α, cases β,
+  dsimp at w,
```

Comment:
Can these two `dsimp`s be merged into one?

Reply:
Fixed.